src/ripple/basics/impl/make_SSLContext.cpp

```diff
@@ -154,16 +163,28 @@ static void x509_sign (X509* x509, EVP_PKEY* evp_pkey)
         LogicError ("X509_sign failed");
 }
 
-static void ssl_ctx_use_certificate (SSL_CTX* const ctx, x509_ptr& cert)
+static void ssl_ctx_use_certificate (SSL_CTX* const ctx, x509_ptr cert)
 {
-    if (SSL_CTX_use_certificate (ctx, cert.release()) <= 0)
+    if (SSL_CTX_use_certificate (ctx, cert.get()) <= 0)
         LogicError ("SSL_CTX_use_certificate failed");
+    {
+        static std::mutex m;
+        static x509_ptr cleanupMem;
+        std::lock_guard<std::mutex> l{m};
```

Comment:
You jumped on one of my quibbles! :-)

I was thinking why not:
```
static x509_ptr cleanupMem = std::move(cert);;
```
But I need to review my suggestion with fresh eyes in the morning before I'm confident about it.